org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService.xtend

```diff
@@ -42,8 +54,17 @@ class RenameService implements IRenameService {
 	
 	@Inject Provider<ServerRefactoringIssueAcceptor> issueProvider
 	
+	Function<EObject, String> attributeResolver = SimpleAttributeResolver.newResolver(String, 'name')
+	
 	override rename(WorkspaceManager workspaceManager, RenameParams renameParams, CancelIndicator cancelIndicator) {
-		val uri = renameParams.textDocument.uri.toUri
+		val textDocument = renameParams.textDocument
+		val uri = textDocument.uri.toUri
+		val positionParams = new TextDocumentPositionParams(new TextDocumentIdentifier(textDocument.uri), renameParams.position)
+		val rangeToRename = prepareRename(workspaceManager, positionParams, cancelIndicator)
+		if (rangeToRename === null|| rangeToRename.getLeft === null || !rangeToRename.getLeft.containsPosition(renameParams.position)) {
+			return null
```

Comment:
Should we also add an issue to the issueAcceptor?

Reply:
This was done here: https://github.com/eclipse/xtext-core/pull/1051/files#diff-2a69e2e40efb5ed1e922cb1546ab8cecR104